### ISSUE TITLE
Wordnet gloss parsing issues

### ISSUE THREAD

**Issue by alvations** [nltk/nltk#2265]
From https://github.com/nltk/nltk/blob/develop/nltk/corpus/reader/wordnet.py#L1396, it looks like the line parsing wasn't done correctly for the example vs the definition of the Synset gloss:

```python
	     # parse out the definitions and examples from the gloss
            columns_str, gloss = data_file_line.split('|')
            gloss = gloss.strip()
            definitions = []
            for gloss_part in gloss.split(';'):
                gloss_part = gloss_part.strip()
                if gloss_part.startswith('"'):
                    synset._examples.append(gloss_part.strip('"'))
                else:
                    definitions.append(gloss_part)
            synset._definition = '; '.join(definitions)
```

As a sanity check, from

```python
from nltk.corpus import wordnet

for ss in wordnet.all_synsets():
    if '"' in ss.definition():
        print(ss, ss.definition())
```

there are these problematic ones:

```
Synset('long.s.09') having or being more than normal or necessary:"long on brains"
Synset('unobjectionable.s.03') not objectionable; it's the means that one can't accept"
Synset('faulty.s.02') characterized by errors; not agreeing with a model or not following established rules; the wrong side of the road"
Synset('dead.s.17') devoid of activity; nothing ever happens here"
Synset('active.a.10') expressing that the subject of the sentence has the semantic function of actor: "Hemingway favors active constructions"
Synset('intercalary.s.01') having a day or month inserted to make the calendar year correspond to the solar year: "Feb. 29 is an intercalary day"
Synset('aggressive.a.01') having or showing determination and energetic pursuit of your ends; positive in his convictions"
Synset('slain.s.01') killed; `slain' is formal or literary as in "slain warriors"
Synset('out_of_play.s.01') (of a ball) "a ball that is out of play is dead"
Synset('fast_asleep.s.01') sleeping deeply; he's sound asleep"
Synset('bashful.s.01') self-consciously timid; lowering my head, I looked at the wall"- Ezra Pound
Synset('gutsy.a.01') marked by courage and determination in the face of difficulties or danger; robust and uninhibited; it was a gutsy thing to do"
Synset('cautious.a.01') showing careful forethought; never making swift decisions"
Synset('certificated.s.01') furnished with or authorized by a certificate: "certificated teachers"
Synset('ineluctable.s.01') impossible to avoid or evade:"inescapable conclusion"
Synset('fixed.s.04') incapable of being changed or moved or undone; e.g. "frozen prices"
Synset('irritating.s.03') causing physical discomfort; they can be very painful"
Synset('crisp.s.06') brief and to the point; effectively cut short; `yes'"
Synset('coordinating.a.01') serving to connect two grammatical constituents of identical construction; and so is `or' in `will you go or stay?'"
Synset('adjacent.s.02') having a common boundary or edge; abutting; touching; Massachusetts and Conncecticut"
Synset('received.s.02') widely accepted as true or worthy; surveys show otherwise"- Economist
Synset('thoughtless.s.02') without care or thought for others; `Let them eat cake'"
Synset('dimensional.s.02') having dimension--the quality or character or stature proper to a person; he is pasty, bland, faceless"- Norman Cousins
Synset('made.s.01') produced by a manufacturing process; rope and nails"
Synset('original.s.04') not derived or copied or translated from something else; not an adaptation"
Synset('hard-to-please.s.01') (of persons) "his father was a hard-to-please taskmaster"
Synset('erratic.s.03') likely to perform unpredictably; sometimes it would start and sometimes it wouldn't"
Synset('commutative.s.01') (of a binary operation) independent of order; as in e.g. "a x b = b x a"
Synset('unselective.s.01') not selective or discriminating; her choices seemed completely random"
Synset('dry.s.16') practicing complete abstinence from alcoholic beverages; I happen to be teetotal"
Synset('humdrum.s.02') tediously repetitious or lacking in variety; all work and no play"
Synset('outside.s.03') originating or belonging beyond some bounds:"the outside world"
Synset('weird.s.02') strikingly odd or unusual; some weird effect of shadow"- Bram Stoker
Synset('adamant.s.01') impervious to pleas, persuasion, requests, reason; she would have none of him"- W.Churchill
Synset('domesticated.s.02') accustomed to home life; others find gratification in it"
Synset('former.a.01') referring to the first of two things or persons mentioned (or the earlier one or ones of several); I prefer the former version to the latter one"
Synset('undrained.a.01') not drained; keep them undrained"
Synset('structural.s.03') affecting or involved in structure or construction; not ornamental elements"
Synset('dry.a.07') without a mucous or watery discharge; a small child with a dry nose"
Synset('divine.s.03') being or having the nature of a god; 'tis God-like to create"-J.G.Saxe
Synset('mundane.s.03') belonging to this earth or world; not ideal or heavenly; yet a mundane wonder of unimagined kind"
Synset('doughy.s.01') having the consistency of dough because of insufficient leavening or improper cooking; it's a doughy mess"
Synset('burdensome.s.01') not easily borne; wearing; I only had to greet the guests"
Synset('fooling.s.01') characterized by a feeling of irresponsibility; it is no fooling matter"
Synset('heedless.a.01') marked by or paying little heed or attention; we know now that it is bad economics"--Franklin D. Roosevelt
Synset('inhumane.a.01') lacking and reflecting lack of pity or compassion; this explains much of the misery and suffering in the world"
Synset('humorless.a.01') lacking humor; a wink of warning"- Truman Capote
Synset('infectious.a.02') easily spread; children catch it from their elders"- Bertrand Russell
Synset('apathetic.s.02') marked by a lack of interest; it is simply indifferent"
Synset('enormous.s.01') extraordinarily large in size or extent or amount or power or degree; that a whole civilization should be dependent on technology"- Walter Lippman
Synset('lowercase.a.01') relating to small (not capitalized) letters that were kept in the lower half of a compositor's type case; a and b and c etc"
Synset('uppercase.a.01') relating to capital letters which were kept in the top half of a compositor's type case; X and Y and Z etc"
Synset('meaty.a.01') like or containing meat; let's have a meaty meal"
Synset('fake.s.02') not genuine or real; being an imitation of the genuine article; it's real synthetic fur"
Synset('obedient.a.01') dutifully complying with the commands or instructions of those in authority; the refractory remain unburdened"- Edmund Burke
Synset('three-piece.a.01') (of clothing) made in or consisting of three parts or pieces; jacket and trousers and vest"
Synset('laid.s.01') set down according to a plan:"a carefully laid table with places set for four people"
Synset('impotent.a.01') lacking power or ability; morality without technology is impotent"- Freeman J.Dyson
Synset('attendant.s.01') being present (at meeting or event etc.) "attendant members of the congreation"
Synset('insular.s.02') suggestive of the isolated life of an island; so deeply private as to seem inaccessible to the scrutiny of a novelist"- Leonard Michaels
Synset('skin-deep.s.01') penetrating no deeper than the skin: "her beauty is only skin-deep"
Synset('for_sure.s.01') not open to doubt; that was for sure"
Synset('real.a.01') being or occurring in fact or actuality; having verified existence; not illusory; not ghosts"
Synset('proper.s.02') having all the qualities typical of the thing specified; not just a snack"
Synset('treated.a.02') given medical care or treatment; if left untreated it lasts two weeks"
Synset('infirm.s.02') lacking firmness of will or character or purpose; give me the daggers" - Shakespeare
Synset('flimsy.s.03') lacking substance or significance; a fragile claim to fame"
Synset('double.s.01') having more than one decidedly dissimilar aspects or qualities; public preaching and private influence"- R.W.Emerson
Synset('unsized.a.02') not fashioned to sizes; one size fits all"
Synset('antisocial.s.01') shunning contact with others; just shy"
Synset('solved.a.01') explained or answered; problems resolved and unresolved"
Synset('unacceptable.s.04') not conforming to standard usage; `You can access your cash at any of 300 automatic tellers'"
Synset('ocular.s.03') visible; give me the ocular proof"- Shakespeare
Synset('well.a.01') in good health especially after having suffered illness or injury; at least I feel well"
Synset('auricular.a.02') relating to or perceived by or shaped like the organ of hearing; I must hear what I read"- George Santayana
Synset('igneous.a.01') produced under conditions involving intense heat; especially from molten magma"
Synset('functional.a.03') relating to or based on function especially as opposed to structure; it is a functional one"
Synset('canonist.a.01') pertaining to or characteristic of a body of rules and principles accepted as axiomatic; e.g. "canonist communism"
Synset('classificatory.a.01') relating to or involving classification:"classificatory criteria"
Synset('vietnamese.a.01') of or relating to or characteristic of Vietnam or its people or its language; the Vietnamese tones"
Synset('sometimes.r.01') on certain occasions or in certain cases but not always; at other times for six months"
Synset('anyhow.r.01') used to indicate that a statement explains or supports a previous statement; anyhow, they're quiet"; anyway, it's gone"; in any case, there was a brief scuffle"
Synset('as_it_is.r.01') in the actual state of affairs and often contrary to expectations; as it is he was severely injured"
Synset('only.r.05') except that; only this time she came out better"
Synset('however.r.02') by contrast; on the other hand; the second, however, took hours"
Synset('besides.r.01') making an additional point; anyway; besides, we can't afford it"
Synset('furthermore.r.01') in addition; furthermore, their quality is improving"; moreover, mice nested there"
Synset('further.r.02') in addition or furthermore; further, they should be here already"
Synset('rather.r.01') on the contrary; rather (or instead), he wrote her a letter"
Synset('enough.r.01') as much as necessary; (`plenty' is nonstandard) "I've had plenty, thanks"
Synset('contrarily.r.02') contrary to expectations; on the contrary, he went out with his friends"
Synset('fearfully.r.01') in fear, "she hurried down the stairs fearfully"
Synset('testily.r.01') in a petulant manner; `Go away!'"
Synset('arrogantly.r.01') in an arrogant manner; afraid of her husband and yet arrogantly proud that she had a husband strong and fierce enough to make her afraid"
Synset('boorishly.r.01') like a boor, "he behaved boorishly at the party"
Synset('indecisively.r.02') without finality; inconclusively; neither side had clearly won but neither side admitted defeat"
Synset('ecologically.r.01') with respect to ecology; economically, it is a disaster"
Synset('inspirationally.r.01') with inspiration; in an inspiring manner, "he talked inspirationally"
Synset('irreverently.r.02') in an irreverent manner; clergymen had been turned from their cures, and churches irreverently used"
Synset('perchance.r.01') through chance, "To sleep, perchance to dream.."
Synset('singularly.r.01') in a singular manner or to a singular degree; he acquired three wives and fourteen children during his Portuguese embassy alone"
Synset('therewithal.r.01') together with all that; besides; and therewithal remit thy other forfeits"- Shakespeare
Synset('untying.n.01') loosening the ties that fasten something; the untying is easy"
Synset('substitution.n.02') the act of putting one thing or person in the place of another: "he sent Smith in for Jones but the substitution came too late to help"
Synset('stride.n.03') significant progress (especially in the phrase "make strides")
Synset('putt.n.01') hitting a golf ball that is on the green using a putter; he didn't sink a single putt over three feet"
Synset('behalf.n.01') as the agent of or on someone's part (usually expressed as "on behalf of" rather than "in behalf of"); 
Synset('duty.n.01') the social force that binds you to the courses of action demanded by that force; every opportunity, an obligation; every possession, a duty"- John D.Rockefeller Jr
Synset('mobilization.n.01') act of assembling and putting into readiness for war or other emergency: "mobilization of the troops"
Synset('carrot.n.04') promise of reward as in "carrot and stick"; 
Synset('vengeance.n.01') the act of taking revenge (harming someone in retaliation for something harmful that they have done) especially in the next life; I will repay, saith the Lord"--Romans 12:19
Synset('cow.n.01') female of domestic cattle: "`moo-cow' is a child's term"
Synset('job.n.03') a workplace; as in the expression "on the job"; 
Synset('worsted.n.01') a woolen fabric with a hard textured surface and no nap; woven of worsted yarns "he wore a worsted suit"
Synset('adequacy.n.01') the quality of being able to meet a need satisfactorily: "he questioned the adequacy of the usual sentimental interpretation of the Golden Rule"
Synset('quaintness.n.01') the quality of being quaint and old-fashioned; its quaintness was appealing"
Synset('scope.n.01') an area in which something acts or operates or has power or control: "the range of a supersonic jet"
Synset('gamut.n.01') a complete extent or range: "a face that expressed a gamut of emotions"
Synset('evil.n.02') that which causes harm or destruction or misfortune; the good is oft interred with their bones"- Shakespeare
Synset('right.n.01') an abstract idea of that which is due to a person or governmental body by law or tradition or nature; it is something that nobody can take away"
Synset('writer's_block.n.01') an inability to write; the words wouldn't come"
Synset('conversion.n.02') a change in the units or form of an expression: "conversion from Fahrenheit to Centigrade"
Synset('exception.n.02') an instance that does not conform to a rule or generalization; the only exception was her last child"
Synset('attractive_nuisance.n.01') anything on your premises that might attract children into danger or harm; they should fence it in"
Synset('roundup.n.02') a summary list; as in e.g. "a news roundup"
Synset('capital.n.04') one of the large alphabetic characters used as the first letter in writing or printing proper names and sometimes for emphasis; capitals were kept in the upper half of the type case and so became known as upper-case letters"
Synset('indirect_discourse.n.01') a report of a discourse in which deictic terms are modified appropriately (e.g., "he said `I am a fool' would be modified to `he said he is a fool'")
Synset('direct_discourse.n.01') a report of the exact words used in a discourse (e.g., "he said `I am a fool'")
Synset('call.n.04') a demand especially in the phrase "the call of duty"
Synset('unknown_quantity.n.01') a factor in a given situation whose bearing and importance is not apparent; he's still an unknown quantity"
Synset('post_office.n.01') a local branch where postal services are available"
Synset('conferee.n.01') a person on whom something is bestowed; the conferees were..."
Synset('dribbler.n.02') a person who dribbles; he needs a bib"
Synset('fast_buck.n.01') quick or easy earnings, "they are traders out to make a fast buck"
Synset('ultimacy.n.01') the state or degree of being ultimate; the final or most extreme in degree or size or time or distance, "the ultimacy of these social values"
Synset('oppression.n.02') the state of being kept down by unjust use of force or authority: "after years of oppression they finally revolted"
Synset('lap_of_the_gods.n.01') beyond human control or responsibility; it's in the lap of the gods now"
Synset('diversification.n.02') the condition of being varied; every day it is the same"
Synset('chip_away.v.01') remove or withdraw gradually: "These new customs are chipping away at the quality of life"
Synset('limit.v.02') restrict or confine, "I limit you to two visits to the pub a day"
Synset('puff.v.07') to swell or cause to enlarge, "Her faced puffed up from the drugs"
Synset('focus.v.05') put (an image) into focus; we cannot enjoy the movie"
Synset('come.v.06') be found or available; The furniture comes unassembled"
Synset('switch.v.04') make a shift in or exchange of; then we switched"
Synset('complete.v.01') come or bring to a finish or an end; others finished in over 4 hours"
Synset('study.v.05') learn by reading books; I must hit the books now"
Synset('draw_out.v.01') cause to speak, "Can you draw her out--she is always so quiet"
Synset('threaten.v.02') to utter intentions of injury or punishment against:"He threatened me when I tried to call the police"
Synset('threaten.v.03') to be a menacing indication of something:"The clouds threaten rain"
Synset('drop.v.06') utter with seeming casualness; drop names"
Synset('decline.v.07') inflect for number, gender, case, etc., "in many languages, speakers decline nouns, pronouns, and adjectives"
Synset('notate.v.01') put into notation, as of music or choreography; in the old days, the steps had to be memorized"
Synset('compromise.v.01') make a compromise; arrive at a compromise; we all must compromise"
Synset('sign_up.v.02') join a club, an activity, etc. with the intention to join or participate, "Sign up for yoga classes"
Synset('beset.v.02') assail or attack on all sides: "The zebra was beset by leopards"
Synset('live_out.v.02') work in a house where one does not live; he can easily commute from his home"
Synset('pleat.v.02') fold into pleats, "Pleat the cloth"
Synset('knock.v.01') deliver a sharp blow or push :"He knocked the glass clear across the room"
Synset('prepose.v.01') place before another constituent in the sentence; Japanese postposes them"
Synset('strew.v.01') spread by scattering ("straw" is archaic)
Synset('dribble.v.03') propel, "Carry the ball"
Synset('croquet.v.01') drive away by hitting with one's ball, "croquet the opponent's ball"
Synset('hitch.v.05') connect to a vehicle: "hitch the trailer to the car"
Synset('bear.v.05') bring forth, "The apple tree bore delicious apples this year"
Synset('manufacture.v.01') put together out of artificial or natural components or parts; He manufactured a popular cereal"
Synset('recite.v.03') render verbally, "recite a poem"
Synset('conduct.v.02') lead, as in the performance of a composition; Barenboim conducted the Chicago symphony for years"
Synset('stay.v.02') stay put (in a certain place); we are not moving to Cincinnati"
Synset('wallow.v.02') roll around, "pigs were wallowing in the mud"
Synset('push.v.01') move with force, "He pushed the table into a corner"
Synset('arouse.v.06') to begin moving, "As the thunder started the sleeping children began to stir"
Synset('crawl.v.05') swim by doing the crawl; they often don't know how to crawl"
Synset('ascend.v.01') travel up, "We ascended the mountain"
Synset('sink.v.04') go under, "The raft sank and its occupants drowned"
Synset('roll_up.v.03') arrive in a vehicle: "He rolled up in a black Mercedes"
Synset('reach.v.07') reach a goal, e.g., "make the first team"
Synset('feel.v.06') undergo passive experience of:"We felt the effects of inflation"
Synset('refocus.v.01') focus once again; The physicist refocused the light beam"
Synset('trust.v.06') extend credit to; I won't pay her debts anymore"
Synset('baby-sit.v.02') work or act as a baby-sitter; I have too much homework to do"
Synset('imprison.v.02') confine as if in a prison; he does not let them go out without a chaperone"
Synset('coerce.v.01') to cause to do through pressure or necessity, by physical, moral or intellectual means :"She forced him to take a job in the city"
Synset('form.v.02') to compose or represent:"This wall forms the background of the stage setting"
Synset('keep.v.01') keep in a certain state, position, or activity; e.g., "keep clean"
Synset('enclose.v.02') close in; darkness enclosed him"
Synset('be.v.05') happen, occur, take place; this was during the visit to my parents' house"
```

**Comment by alvations** [comment]
These regexes might work better:

```python
definition = re.sub(r"[\"].*?[\"]", "", gloss)
examples = re.findall(r'"([^"]*)"', gloss)
```

E.g.

```python

from lazyme import find_files

for filename in find_files('/Users/liling.tan/nltk_data/corpora/wordnet/', 'data.*'):
    with open(filename) as fin:
        for i, line in enumerate(fin):
            if i > 100:
                break
            if line.startswith(' '):
                continue
            columns_str, gloss = line.strip().split('|')
            definition = re.sub(r"[\"].*?[\"]", "", gloss)
            examples = re.findall(r'"([^"]*)"', gloss)
            print(definition.strip(' ;,'))
            print(examples)
            print()

```

**Comment by alvations** [comment]
BTW, regarding the line: https://github.com/nltk/nltk/blob/develop/nltk/corpus/reader/wordnet.py#L1460 

```python
tups = lemma_pointers[source_lemma_name, symbol]
tups.append((pos, offset, target_index))
```

I think it is also not necessary to first load/retrieve the `defaultdict(list)` key, should be able to do this directly:

```python
lemma_pointers[source_lemma_name, symbol].append((pos, offset, target_index))
```

**Comment by alvations** [comment]
Patched in #2294 =)

**Comment by multimeric** [comment]
I feel like this is still an issue. If I run your example above, I get:

```
Synset('assimilating.s.01') capable of taking (gas, light, or liquids) into a solution; "an assimilative substance
Synset('faulty.s.02') characterized by errors; not agreeing with a model or not following established rules; ; ; the wrong side of the road"
Synset('classical.s.05') of or pertaining to or characteristic of the ancient Greek and Roman cultures; ; "classical
Synset('toed.a.01') having a toe or toes of a specified kind; often used in combination; five-toed"
Synset('wood-fired.s.01') fueled by wood; "a wood-burning stove'
Synset('flimsy.s.03') lacking substance or significance; ; ; ; a fragile claim to fame"
Synset('compound.a.01') composed of more than one part; compound flower heads"
Synset('classical.s.04') (language) having the form used by ancient standard authors; "classical Greek
Synset('paramedical.a.01') of or denoting a person who assists physicians and nurses or is trained physicians and nurses in their activities; ambulance drivers are paramedical personnel"
Synset('vietnamese.a.01') of or relating to or characteristic of Vietnam or its people or its language ; ; the Vietnamese tonesVietnamese boat people"
Synset('shopping.n.02') the commodities purchased from stores; women carrying home shopping didn't give me a second glance"
Synset('notice.n.01') an announcement containing information about an event; ; ; "a notice of sale
Synset('post_office.n.01') a local branch where postal services are available"
Synset('bondage.n.01') the state of being under the control of a force or influence or abstract power; he sought release from his bondage to Satana self freed from the bondage of time"
Synset('drop.v.06') utter with seeming casualness; ; drop names"
Synset('side.v.01') take sides for or against; ; m siding against the current candidate"
Synset('manufacture.v.01') put together out of artificial or natural components or parts; ; ; He manufactured a popular cereal"
Synset('ascend.v.08') move to a better position in life or to a better job; "She ascended from a life of poverty to one of great
Synset('refocus.v.01') focus once again; The physicist refocused the light beam"
Synset('enclose.v.02') close in; darkness enclosed him"
```
